### PR TITLE
Using error notification when code errors

### PR DESCRIPTION
**Pull request by mat-mcloughlin**
Currently brackets only shows the warning style notify icon when there the `CodeInspector` finds problems with the code. This change means it will now show the error icon if any of the errors have the `error.type = Type.ERROR`

Not the worlds biggest fan of the icon but it was already present. If you'd like it changing for a different one let me know. 

image

This is particularly useful for my Omnisharp plugin :)

### CODE REVIEW

**Comment by ingorichter** [comment]
@mat-mcloughlin are you still interested in getting this PR finished? It seems that there are only minor issues preventing this from being merged. Just saying ;-)

**Comment by mat-mcloughlin** [comment]
@ingorichter hey, sorry I've been very AFK recently with work. I'll sort it this week

**Comment by ingorichter** [comment]
No worries, I just wanted to make sure that this doesn't fall off the plate. :-)

**Comment by mat-mcloughlin** [comment]
Changed the tests.

File: src/language/CodeInspection.js [hunk 1]
```diff
@@ -272,7 +272,7 @@ define(function (require, exports, module) {
      * @param {Array.<{name:string, scanFileAsync:?function(string, string):!{$.Promise}, scanFile:?function(string, string):Object}>} providersReportingProblems - providers that reported problems
      * @param {boolean} aborted - true if any provider returned a result with the 'aborted' flag set
```
Comment (redmunds) [inline src/language/CodeInspection.js]: Still need to update annotation. Also should update description about updating icon. Something like:

```js
    /**
     * Update the title of the problem panel, the tooltip of the status bar indicator, and the icon of
     * the status bar indicator. The title and the tooltip will change based on the number of problems
     * reported and how many provider reported problems. The icon changes for warnings or errors.
     * 
     * @param {Number} numProblems - total number of problems across all providers
     * @param {Array.<{name:string, scanFileAsync:?function(string, string):!{$.Promise}, scanFile:?function(string, string):Object}>} providersReportingProblems - providers that reported problems
     * @param {boolean} hasError - true if at least 1 problem is an error
     * @param {boolean} aborted - true if any provider returned a result with the 'aborted' flag set
     */
```

**Comment by abose** [comment]
I do not have context for this issue,
@mat-mcloughlin Why was this closed?

**Pull request closed without merging**